### ISSUE TITLE
Your Redis instance may be publicly accessible.

### ISSUE THREAD

**Issue by fshmcallister** [WoTTsecurity/wott-io#214]
We detected that a {service} instance on {devices} may be accessible remotely. Consider either blocking port {port} through the WoTT firewall management tool, or re-configure {service} to only listen on localhost. Part of #198

**Comment by fshmcallister** [comment]
**TITLE**
A Redis instance may be publicly accessible 

**TL;DR**
Your Redis instance may be publicly accessible. Consider reconfiguring your Redis to listen only on the localhost, or using secured channels only such as TSP through port 6379.

**FAQ**
Redis (remote dictionary server) is an open-source data structure store with several uses, including as an alternative database system to SQL variants.

**Comment by vpetersson** [comment]
> No-SQL database system

NoSQL

> over an internet network

?

**Comment by fshmcallister** [comment]
As in accessible via internet as opposed to a private connection. Does it not make sense?

**Comment by vpetersson** [comment]
I would probably say "exposed over the internet" instead for greater clarity.

**Comment by fshmcallister** [comment]
Sure- let me update the block now

**Comment by fshmcallister** [comment]
Done

**Comment by fshmcallister** [comment]
**TITLE**
A Redis instance may be publicly accessible 

**TL;DR**
Your Redis instance may be publicly accessible. Consider reconfiguring your Redis to listen only on the localhost, or using secured channels only such as TCP through port 6379.

**FAQ**
Redis (remote dictionary server) is an open-source data structure store with several uses, including as an alternative NoSQL database system. However, Redis does not have any robust security features of its own, and is intended for use by trusted clients in a trusted network. It is therefore not recommended to have a Redis instance exposed over the internet, even across a TCP channel through port 6379. To circumvent this, set your Redis instance to only listen to the localhost by binding it to the localhost. You can do this through your firewall directly, for example through WoTT's firewall tool, or the service config itself.

Through the Redis config file, \`/etc/redis/redis.conf\`, you will already see that the line binding the server to the localhost exists, and just needs to be uncommented:

\`\`\`
bind 127.0.0.1
\`\`\`
By default, the service listens to the localhost. This may have changed if you have since changed the Redis settings. If there are specific IPs you wish to add, you will need to add the address to the config through the same \`bind\` command. We would strong recommend enabling password protection as well in this instance by adding this in the config:

\`\`\`
requirepass yourPasswordHere
\`\`\`

**Comment by a-martynovich** [comment]
@fshmcallister subtitle?
